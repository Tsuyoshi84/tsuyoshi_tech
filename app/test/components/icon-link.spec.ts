import { mount } from '@vue/test-utils'
import IconLink from '@/components/icon-link.vue'

const makeWrapper = () => {
  return mount(IconLink, {
    propsData: { alt: 'Example', href: 'https://example.com', color: 'red' },
    slots: {
      default: '<div>slot-content</div>',
    },
    stubs: {},
  })
}

describe('IconLink', () => {
  test('is a Vue instance', () => {
    const wrapper = makeWrapper()
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  describe('when not hover', () => {
    test('should use default color', (done) => {
      const wrapper = makeWrapper()
      expect(wrapper.html()).toMatchSnapshot()
      done()
    })
  })

  describe('when hover', () => {
    test('should use given color', (done) => {
      const wrapper = makeWrapper()
      wrapper.trigger('mouseover')
      wrapper.vm.$nextTick(() => {
        expect(wrapper.html()).toMatchSnapshot()
        done()
      })
    })
  })
})
