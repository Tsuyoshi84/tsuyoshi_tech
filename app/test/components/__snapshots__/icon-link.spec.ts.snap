// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconLink when hover should use given color 1`] = `
<div class="box"><a href="https://example.com" alt="Example" aria-label="Example" class="link-icon" style="color: red;">
    <div>slot-content</div>
  </a></div>
`;

exports[`IconLink when not hover should use default color 1`] = `
<div class="box"><a href="https://example.com" alt="Example" aria-label="Example" class="link-icon" style="color: currentColor;">
    <div>slot-content</div>
  </a></div>
`;
